File: 04.Queue/Data structure Queue.cpp
#include <iostream>
// queue header include
#include <queue>
using namespace std;

int main()
{
    // Queue declaration and creation
    queue<int> que;
    
    que.push(1); // 1 added
    que.push(2); // 2 added
    que.push(3); // 3 added
    cout << que.front() << "\n"; // Front data verification : 1
    
    que.push(4); // 4 added
    que.push(5); // 5 added
    cout << que.front() << "\n"; // Front data verification : 1

    que.pop(); // Front data removal : 1 remove
    cout << que.front() << "\n"; // Front data verification : 2
    
    // Removes all data in the queue and print
    while(!que.empty())
    {
        cout << que.front() << "\n";
        que.pop();
    }    
}

File: 04.Queue/Data structure Queue.java
import java.util.*;

public class data structure Queue {
    public static void main(String[] args) {
        
        // Queue declaration and creation
        Queue<Integer> queue = new ArrayDeque<Integer>();
        
        queue.offer(1); // 1 added
        queue.offer(2); // 2 added
        queue.offer(3); // 3 added
        System.out.println(queue.peek()); // Front data verification : 1
        
        queue.offer(4); // 4 added
        queue.offer(5); // 5 added
        System.out.println(queue.peek()); // Front data verification : 1

        System.out.println(queue.poll()); // Front data verification and removal : 1
        System.out.println(queue.peek()); // Front data verification : 2
        
        // Removes all data in the queue and print
        while(!queue.isEmpty())
        {
            System.out.println(queue.poll());
        }
    }
}